provider/datagen/src/transform/cldr/datetime/neo.rs

```diff
@@ -582,6 +582,11 @@ fn apply_numeric_overrides(lp: &ca::LengthPattern, pattern: &mut pattern::runtim
         if let pattern::PatternItem::Field(ref mut field) = *item {
             // only replace numeric items
             if field.length != FieldLength::One {
+                assert!(
+                    field.length != FieldLength::TwoDigit || symbol_to_replace.is_some(),
```

Comment:
I'd prefer the condition to be stricter

```suggestion
                    field.length != FieldLength::TwoDigit || symbol_to_replace != Some(symbol),
```